server/task_scheduler.go

```diff
@@ -233,17 +233,28 @@ func (s *TaskScheduler) Register(taskType string, executor TaskExecutor) {
 
 // ScheduleIfNeeded schedules the task if its required check does not contain error in the latest run
 func (s *TaskScheduler) ScheduleIfNeeded(ctx context.Context, task *api.Task) (*api.Task, error) {
-	// For now, only schema update task has required task check
+	// every task should pass timing task check
```

Comment:
IIUC,  we would never pass the task check if task.NotBeforeTs == 0. In this case, we won't run the check, and since we won't run the check, passCheck always returns false.

Reply:
My fault, forget to add circuit logic to the scheduler